test/integration/authorizeUsersAndService.test.ts

```diff
@@ -89,7 +89,7 @@ describe("Middleware | Authorization", function () {
       return res.json({ message: "pong" });
     };
 
-    router.get("/for-super-user", authorization([ROLES.SUPERUSER], [Services.CRON_JOB_HANDLER]), pongHandler);
+    router.get("/for-super-user", authorizeAndAuthenticate ([ROLES.SUPERUSER], [Services.CRON_JOB_HANDLER]), pongHandler);
```

Comment:
Please add more tests like, if only the super user role is set, it should not be accessible by other service, and vice versa, 
See if you can cover some more tests case here

Reply:
> if only the super user role is set, it should not be accessible by other service

This middleware is created to make sure that services would also be able to work as a superuser without cookies.

The purpose of **authorizeAndAuthenticate** is very specific to the cases where we need routes to support both the cookie based [ superuser or otherwise ] and the service authentication/authorization, for which test cases are covered

Reply:
The integration cases have been covered here:
https://github.com/Real-Dev-Squad/website-backend/blob/07d1129f180ce9a112b627fad25bc0bd15345ab2/test/integration/authorizeUsersAndService.test.ts